docs/userGuide/syntax/questions.mbdf

```diff
@@ -1,130 +1,325 @@
-## Questions
+## Questions and Quizzes
 
-**Question component consists of a question body, a hint and an answer.**
+Question and quiz components provide an easy way to test readers on the relevant content topic in the page.
 
-<include src="codeAndOutput.md" boilerplate >
-<variable name="highlightStyle">html</variable>
-<variable name="code">
-<question>
-  The question body. <md>:grey_question:</md>
-  <div slot="hint">
-    Question hint. <md>:crystal_ball:</md>
-  </div>
-  <div slot="answer">
-    Question answer. <md>:pencil:</md>
-  </div>
+#### Introduction
+
+Question components (`<question>`) can be one of the following types: **Mcq**, **Checkbox** or **Text**.
+
+<div class="px-5">
+<question type="checkbox" header="Which of the following is true? (**Card Header**)" hint="This is the question hint, you can use **markdown** here as well.">
+
+Anything you place inside a question not in a slot is inserted into the **card body**!
+
+<q-option reason="We aren't concatenating strings!">
+  20 + 20 = 2020
+</q-option>
+<q-option reason="Division by zero is **undefined**!">
+  1 / 0 = infinity
+</q-option>
+<q-option correct>
+11 / 11 = 1
+</q-option>
 </question>
-</variable>
-</include>
+</div>
+
+
+In all cases, note that content directly inserted in between `<question>...</question>` will be inserted into the card's body.
 
+To output your markup to the header, or insert content into the hint tip box, simply insert your markup inside the `header` and `hint` attributes respectively:
+```html
+<question header="Which of the following is true? (**Card Header**)"
+          hint="This is the question hint, you can use **markdown** here as well.">
+  ...
+</question>
+```
+
+Click the hint button above to see how the hint box turns out!
+
+<box type="tip" seamless>
+
+Placing the question into the card header is optional. You may also wish to include the question directly in the card body, omitting the header entirely.
+</box>
 
-**If you leave the hint and answer `<div>` blank, a default hint and answer will be provided instead.**
+****Header and Hint slots****
+
+If you require more expressive formatting for your header or hint markup, simply use the `<template slot="header">` and `<template slot="hint">` slots like so:
 
 <include src="codeAndOutput.md" boilerplate >
 <variable name="highlightStyle">html</variable>
 <variable name="code">
-<question>
-  This question has no hint or answer.
-  <div slot="hint"></div>
-  <div slot="answer"></div>
+<question type="text">
+  <template slot="header">
+
+  Which of the following is true? (**Card Header**)
+
+  Paragraph **2** :pizza:
+
+  </template>
+  <template slot="hint">
+
+  This is the question hint, you can use **markdown** here as well.
+
+  Paragraph **2**
+
+  </template>
 </question>
 </variable>
 </include>
 
```

Comment:
It's harder but better if the example used is a real question that illustrates all the points.

Reply:
dosen't this defeat the purpose of having an introduction section (not having touched on text / mcq / checkbox yet)? chucking it all into mcq and checkbox questions seems a little heavy

Reply:
I truncated the code though, and kept the `q-option`s in the output (since mcq/checkbox hasn't been explained yet), if that's what you meant

Reply:
I meant the example should be a real-word example, not a meta example.
e.g., Instead of `Paragraph **2**` (which will never appear in a real-question), use an answer that actually has two real paragraphs. Coming up with such illustrative real-world examples are harder though. I think some of the examples given match that criterion, but some don't.

Reply:
ahh my bad. will try to find an interesting one 😏